### PR TITLE
Allow custom pass and dbname

### PR DESCRIPTION
**Pull request by pawel-yesware**
This will allow custom MYSQL_PASSWORD and MYSQL_DATABASE_NAME environments.

This is very similar to what we did to POSTGRES plugin.

MYSQL also have root password which we don't expose we only expose mysql user.

@mkostick @justin-yesware

### CODE REVIEW

File: commands [hunk 1]
```diff
@@ -30,9 +30,16 @@ case "$1" in
...
-    password=$(openssl rand -hex 8)
     echo "$rootpassword" > "$SERVICE_ROOT/ROOTPASSWORD"
-    echo "$password" > "$SERVICE_ROOT/PASSWORD"
+
+    if [[ -n $MYSQL_PASSWORD ]]; then
+      password="$MYSQL_PASSWORD"
+      echo "$password" > "$SERVICE_ROOT/PASSWORD"
+    else
+      password=$(openssl rand -hex 8)
+      echo "$password" > "$SERVICE_ROOT/PASSWORD"
```
Comment (justinmills) [inline commands]: These lines can be moved outside of the if block.

File: commands [hunk 2]
```diff
@@ -30,9 +30,16 @@ case "$1" in
     mkdir -p "$SERVICE_ROOT/config" || dokku_log_fail "Unable to create service config directory"
     echo -e "[mysqld]\nperformance_schema = 0" > "$SERVICE_ROOT/config/disable_performance_schema.cnf"
     rootpassword=$(openssl rand -hex 8)
-    password=$(openssl rand -hex 8)
     echo "$rootpassword" > "$SERVICE_ROOT/ROOTPASSWORD"
```
Comment (justinmills) [inline commands:33]: We actually use the root user for mysql. That said, only a few of us use it so we could change it pretty easily.

File: functions [hunk 3]
```diff
@@ -271,8 +271,15 @@ service_url() {
   local SERVICE_ROOT="$PLUGIN_DATA_ROOT/$SERVICE"
 
   local PASSWORD="$(cat "$SERVICE_ROOT/PASSWORD")"
+
+  if [[ -e "$SERVICE_ROOT/DATABASE_NAME" ]];then
```
Comment (justinmills) [inline functions]: whitespace around `;`.

File: commands (hunk 2 above)
Comment (pawel-yesware) [inline commands:33]: Should we set the same password for ROOT and for MYSQL users?

**Comment by justinmills** [comment]
Right now we're connecting as root with no password, but I think to get this working, we could change that.

Couple of comments.

File: commands (hunk 2 above)
Comment (justinmills) [inline commands:33]: I don't think so. But setting root's password via a similar mechanism might not be a bad thing. That said, I'm not sure you could set it to "" (blank) and things would work.

**Comment by pawel-yesware** [comment]
Or we can just disable mysql authentication and connect to mysql without password?

**Comment by pawel-yesware** [comment]
deleting this branch

**Comment by pawel-yesware** [comment]
OK So i have moved the echo outside of if statement and fixed the whitespace.

**Comment by justinmills** [comment]
g2g

**Pull request merged**